lib/engine/game/base.rb

```diff
@@ -1875,6 +1875,11 @@ def ability_right_time?(ability, time, strict_time)
         return false if strict_time && !ability.when
         return true unless time
 
+        current_step = @round.steps.find { |s| s.blocks? && !s.passed? && !s.inactive }
```

Comment:
can you use active_step? and if not maybe add it to the round base

Reply:
`active_step` leads to an infinite loop:

* `Game#active_step` just calls `Round#active_step` https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/game/base.rb#L477-L479
* `Round#active_step` with no entity passed calls `step.blocking?` https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/round/base.rb#L122-L126
* `Step#blocking?` calls `current_actions` https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/step/base.rb#L75-L77
* `Step#current_actions` calls actions` https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/step/base.rb#L57-L61
* In some cases, a step's `actions` calls `Game#abilities`, completing the loop, e.g., `SpecialTrack` calls `tile_lay_abilities` which in turn calls `Game#abilities` https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/step/special_track.rb#L14-L19 https://github.com/tobymao/18xx/blob/d025f57c5679b011a9312136fd899032b3970f91/lib/engine/step/tracker.rb#L41-L43


I'll add it to the round base